src/viper/README.md

```diff
@@ -161,23 +163,121 @@ Below, we summarize the language features that _Motoko-san_ currently supports.
   
     Extending to actor classes and modules is _simple_.
 
-* **Primitive types** — Only integer and Boolean types are supported (including literals of these types):
-  
-  * `Bool`: `not`, `or`, `and`, `implies` (short circuiting semantics)
+* **Types and operations**
+
+    * Type `Bool`
+        * Literals: `true`, `false`
+        * Operations: `not`, `or`, `and`, `implies` (short circuiting semantics)
+
+    * Type `Int`
+        * Literals: `0`, `23`, `-5`, etc
+        * Operations: `+`, `/`, `*`, `-`, `%`
+
+    * Type `Nat`
+        * Supported via translation to `Int` (sound because `Nat` is a
+          subtype of `Int`)
+
+    * Relations on `Bool`, `Int`, `Nat`
+        * `==`, `!=`, `<`, `>`, `>=`, `<=`
+
+    * Array types `[T]` and `[var T]`, where `T` ∈ {`Bool`, `Int`, `Nat`}
+        * Construction: `[]`, `[x]`, `[x, y, z]`, `[var x, y, z]`, etc
+        * Indexing: `a[i]`, can be used on the LHS of an assignment
+        * Operations: `a.size()`
+
+    * Tuple types `(T₁, T₂)`, `(T₁, T₂, T₃)`, etc, where `T` ∈ {`Bool`, `Int`, `Nat`}
+        * Construction: `(a, b)`, `(a, b, c)`, etc
+        * Projections: `x.1`, `x.2`, etc
+        * Deconstruction by pattern matching:
+
+            * `let (a, b) = x`
+            * `switch x { case (a, b) ... }`
+
+          Nested matching is _not_ supported, i.e. subpatterns for tuple
+          elements `a`, `b`, `c` in `(a, b, c)` must all be variable patterns.
+
+    * Option types `Option<T>` (no restrictions on `T`)
+        * Construction: `null`, `?a`
+        * Deconstruction by pattern matching:
+
+          ```motoko
+          switch x {
+              case null ...;
+              case (?a) ...;
+          }
+          ```
+
+          Nested matching is _not_ supported, i.e. the subpattern `a` in `?a`
+          must be a variable pattern.
+
+    * Variant types `{ #A; #B : T₁; #C : (T₂, T₃) }` (with limitations)
+        * Construction: `#A`, `#B(x)`, `#C(x, y)`
+        * Deconstruction by pattern matching:
 
-  * `Int`: `+`, `/`, `*`, `-`, `%`
+          ```motoko
+          switch x {
+              case #A ...;
+              case #B(x) ...;
+              case #C(x, y) ...;
+          }
+          ```
 
-  * Relations: `==`, `!=`, `<`, `>`, `>=`, `<=`
+          Nested matching is _not_ supported, i.e. the subpatterns `x`, `y` in
+          `#B(x)` and `#C(x, y)` must all be variable patterns.
 
-  Supporting `Text`, `Nat`, `Int32`, tuples, record, and variants is _simple_.
+        * Generic and recursive variants are supported, e.g.
 
-    Supporting `Option<T>` types is _trivial_.
+          ```motoko
+          type List<T> = { #Nil; #Cons : (T, List<T>) }
+          ```
+
+        * Limitation 1: Nominative equality. In the Viper translation, equality
```

Comment:
```suggestion
        * Limitation 1: Nominal equality. In the Viper translation, equality
```

Reply:
How silly of me, thank you for the correction.